src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -135,16 +143,46 @@ private void InitializeCore()
                 _oleComponent.ModalStateChanged += OnModalStateChanged;
             }
 
-            UpdateStateMachine();
+            // run it from background and fire and forget
+            StartUpdateStateMachine();
         }
 
-        private void UpdateStateMachine()
+        private void StartUpdateStateMachine()
         {
-            AssertIsForeground();
+            // cancel previous state machine update request
+            _cancellationTokenSource.Cancel();
+            _cancellationTokenSource = new CancellationTokenSource();
 
-            var currentStatus = IsReSharperEnabled();
+            var cancellationToken = _cancellationTokenSource.Token;
+
+            // make sure all state machine change work is serialized so that cancellation
+            // doesn't mess the state up.   
+            _lastTask = _lastTask.ContinueWith(_ =>
```

Comment:
❗️ Use `SafeContinueWithFromAsync`

Reply:
@sharwell, this is fixed in https://github.com/dotnet/roslyn/pull/31918/commits/ab2060e545d70d44d53c7298c99f658d0548c730